### PR TITLE
1729 SHOW TAG VALUES should support keys wrapped in double quotes

### PR DESCRIPTION
**Pull request by rothrock**
I couldn't think of a better way to do it. I'll also need to make a parseIdentListNoQuote if what I've done below seems reasonable.

@dgnorton, @pauldix, @benbjohnson I'd welcome your insight on this if y'all have any time.

Not to be discordant, but I think it's a little odd that we support unquoted strings on the RHS of the 'equals' operator in the first place.

Also, The code change in scanner.go isn't germane; I just couldn't see how that was working correctly.

### CODE REVIEW

**Comment by dgnorton** [comment]
The scanner already supports both quoted and unquoted identifiers.  See...
https://github.com/influxdb/influxdb/blob/master/influxql/scanner.go#L34-L36
https://github.com/influxdb/influxdb/blob/master/influxql/scanner.go#L127-L163

We should add a test for this in scanner_test.go and/or parser_test.go though if there isn't one there already.

This bug isn't in the scanner or parser.  It's in the processing of the query.

File: influxql/parser.go [hunk 1]
```diff
@@ -372,6 +372,17 @@ func (p *Parser) parseIdent() (string, error) {
 	return lit, nil
 }
 
+// parseIdentNoQuote parses an identifier and strips outermost double-quotes if present.
+func (p *Parser) parseIdentNoQuote() (string, error) {
+	tok, pos, lit := p.scanIgnoreWhitespace()
+	if tok != IDENT {
+		return "", newParseError(tokstr(tok, lit), []string{"identifier"}, pos)
+	}
+	lit = strings.TrimSuffix(lit, `"`)
```
Comment (otoolep) [inline influxql/parser.go]: Why not just Trim()?

Comment (dgnorton) [inline influxql/parser.go]: No changes needed in the parser to fix this bug.

**Comment by rothrock** [comment]
Ha, I didn't know issue 1729 was a bug. Looked like an enhancement request, and this pull request embodies a code change that addresses the issue. Moreover, this change narrowly confines itself to quoted identifiers in SHOW TAG VALUES statements.

Re quoted identifiers, the scanner certainly notices double-quotes. The scanIdent function even goes to the trouble to replace them after scanString has extracted the meaningful content originally stored between them. I think that is a problem, but clearly other sections of code have come to expect that behavior. See the quote-replacement logic here:
https://github.com/influxdb/influxdb/blob/master/influxql/scanner.go#L143-145

Since tags can have whitespace in them, wouldn't it be better to insist that the right-hand side of the '=' operator always be quoted? Note that SQL uses single quotes.

Is the idea to support identifiers-as-values the way a CLI shell, like bash, would? Example:
```bash
dogfish:scratch rothrock$ ls log_entry.json 
log_entry.json
dogfish:scratch rothrock$ ls "log_entry.json" 
log_entry.json
```
Maybe @dgnorton or @pauldix can better restate issue 1729?

I'm pretty sure line 140 of scanner.go is not quite right.

**Comment by dgnorton** [comment]
Identifiers are things like measurement names, retention policy names, tag keys, or field names.  The user can avoid quoting them by sticking to a restricted set of chars...or, if they want to use restricted chars, they can wrap the identifiers in double quotes.

Another case where we require double quotes is disambiguation of identifiers.  For example, there are places in the query syntax that allow the user to specify `<db name>.<retention policy>.<measurement>`.  Let's say the user wants to specify database `foo`, retention policy `1h.cpu`, and measurement `cpu`.  Notice that `.` is part of the retention policy name.  If the user were to write this as `foo.1h.cpu.cpu`, there is ambiguity.  To resolve the ambiguity, it must be written `foo."1h.cpu".cpu`.  We also allow this: `"foo".."cpu"`, which means database `foo`, the default retention policy in database foo, and measurement `cpu`.

Single quotes are only for string literals and string literals must always be single quoted.

We can't always require the RHS of `=` to be quoted because not all literals are strings.  InfluxQL also supports numbers, durations, date/time, etc.

InfluxQL also allows identifers and literals to be on either side of the `=` in an expression.  Although most users will put literals on the RHS, it would be awkward and, in my opinion, wrong to force it.

Re line 140, the parser should maintain the original input in the AST.  I.e., the parser should not change what the user said...it should just check it for correctness.  We may want to provide other tools that use the parser that need to know about the quotes (e.g., query formatter).

We maintain a formal (EBNF) definition of InfluxQL syntax here... https://github.com/influxdb/influxdb/blob/master/influxql/INFLUXQL.md

Re the actual issue.  Yeah, I can see how you might have thought it was an enhancement.  Sorry, I should have clarified.  It is a bug and I thought it would be a good learning experience to trace it through the parser into the execution code, which is where the bug is.  Start looking in server.go in `executeShowTagValues` and trace execution of the function from there.  The fix will be a one-liner + some tests.  :)

**Comment by rothrock** [comment]
influxql learnings are over for Mr Rothrock for today. We will subsequently compose a new pull request that addresses this issue in a completely different part of the code base.

**Pull request closed without merging**